src/Router/RouteCollectorDelegator.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mezzio\Router;
+
+use Laminas\ServiceManager\Factory\DelegatorFactoryInterface;
+use Mezzio\MiddlewareFactoryInterface;
+use Psr\Container\ContainerInterface;
+use Webmozart\Assert\Assert;
+
+use function assert;
+use function is_array;
+use function is_string;
+
+final class RouteCollectorDelegator implements DelegatorFactoryInterface
+{
+    /**
+     * RouteCollector Delegator
+     *
+     * Delegates around the RouteCollectorInterface and triggers all registered route providers prior to returning the
+     * RouteCollector instance.
+     *
+     * @inheritDoc
+     */
+    public function __invoke(
+        ContainerInterface $container,
+        $name,
+        callable $callback,
+        ?array $options = null
+    ): RouteCollectorInterface {
+        $collector = $callback();
+        Assert::isInstanceOf($collector, RouteCollectorInterface::class);
+
+        $config = $container->get('config');
+        Assert::isArray($config);
+
+        $routerConfig = $config['router'] ?? [];
+        assert(is_array($routerConfig));
```

Comment:
above at line 36 is using Assert::isArray(), here should be same instead of native assert?

Reply:
No, we want this to crash if config is not an array. I've used native assert on the router config because it _should_ be an array by convention.